polkadot/node/collation-generation/src/lib.rs

```diff
@@ -598,3 +618,37 @@ fn erasure_root(
 	let chunks = polkadot_erasure_coding::obtain_chunks_v1(n_validators, &available_data)?;
 	Ok(polkadot_erasure_coding::branches(&chunks).root())
 }
+
+// Checks if the runtime supports `request_claim_queue` and executes it. Returns `Ok(None)`
+// otherwise. Any [`RuntimeApiError`]s are bubbled up to the caller.
+async fn fetch_claim_queue(
+	sender: &mut impl overseer::CollationGenerationSenderTrait,
+	relay_parent: Hash,
+) -> crate::error::Result<Option<BTreeMap<CoreIndex, VecDeque<ParaId>>>> {
+	if has_required_runtime(
+		sender,
+		relay_parent,
+		RuntimeApiRequest::CLAIM_QUEUE_RUNTIME_REQUIREMENT,
+	)
+	.await
```

Comment:
I see your point but I think it is more or less the same.

Each runtime call has got a version check before it (done by the runtime api subsystem). So in the current case we have got:
Version check (by `has_required_runtime`) -> Version check (by runtime api subsystem) -> Runtime call

If we call the new api directly on an old runtime and handle the error we will have the same:
Version check (by runtime api subsystem for the 'old call') -> Version check (by runtime api subsystem for the 'new call') -> Runtime call

So in terms of runtime calls - it's the same when we have got old runtime.

For a new runtime we do save one version check but (1) we will remove the version check once everything is released on Polkadot and (2) the runtime calls are cached and never hit the runtime.

Considering these - I'd stick with the current version because it is more explicit.

Another drawback of relying on `NotSupported` for version check is that the former happens in other weird cases - e.g. [a runtime call being executed on pruned blocks](https://github.com/paritytech/polkadot-sdk/issues/793). Having `NotSupported` errors polluting the logs won't be ideal especially when we chase bugs.

Reply:
> For a new runtime we do save one version check but (1) we will remove the version check once everything is released on Polkadot and (2) the runtime calls are cached and never hit the runtime.

yes, that's exactly my point, saving one version check for a new runtime. We'd also save one round-trip to the runtime API subsystem, regardless of the runtime version.

anyway, I don't have very strong feelings about this

Reply:
Your point is valid. I've created an issue for this - https://github.com/paritytech/polkadot-sdk/issues/3756